engine/consensus/matching/engine.go

```diff
@@ -376,17 +407,17 @@ func (e *Engine) storeReceipt(receipt *flow.ExecutionReceipt, head *flow.Header,
 	// mempool was wiped during a node crash.
 	err = e.receiptsDB.Store(receipt) // internally de-duplicates
 	if err != nil && !errors.Is(err, storage.ErrAlreadyExists) {
-		return fmt.Errorf("could not persist receipt: %w", err)
+		return false, fmt.Errorf("could not persist receipt: %w", err)
 	}
-	return nil
+	return true, nil
 }
 
 // storeIncorporatedResult creates an `IncorporatedResult` and adds it to incorporated results mempool
 // Error returns:
 // 	* `engine.DuplicatedEntryError` [sentinel error] if entry already present in mempool
 //	* exception in case something went wrong
 // 	* nil in case of success
-func (e *Engine) storeIncorporatedResult(receipt *flow.ExecutionReceipt, log *zerolog.Logger) error {
+func (e *Engine) storeIncorporatedResult(receipt *flow.ExecutionReceipt) (bool, error) {
```

Comment:
as you have changed the behaviour os this method (turned the sentinel into a dedicated return value), please update the goDoc accordingly